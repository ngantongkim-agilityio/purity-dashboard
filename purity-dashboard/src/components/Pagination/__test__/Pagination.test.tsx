import { render } from '@testing-library/react';

// Components
import { Pagination } from '..';

jest.mock('next/navigation', () => ({
  ...jest.requireActual('next/navigation'),
  useSearchParams: jest.fn(
    () =>
      new URLSearchParams({
        query: 'test',
        page: '3'
      })
  ),
  usePathname: jest.fn(),
  useRouter: jest.fn(() => ({ replace: jest.fn() }))
}));

describe('Pagination component', () => {
  it('Should render Pagination component with default props', () => {
    const { container } = render(<Pagination totalPages={3} />);

    expect(container).toMatchSnapshot();
  });
});
