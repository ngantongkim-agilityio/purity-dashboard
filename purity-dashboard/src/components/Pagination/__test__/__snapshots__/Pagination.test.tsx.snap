// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component Should render Pagination component with default props 1`] = `
<div>
  <div
    class="inline-flex"
  >
    <a
      class="flex h-10 w-10 items-center justify-center rounded-md border hover:bg-gray-100 mr-2 md:mr-4"
      href="undefined?query=test&page=2"
    >
      <svg
        aria-hidden="true"
        class="w-4"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.5 19.5 3 12m0 0 7.5-7.5M3 12h18"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </a>
    <div
      class="flex -space-x-px"
    >
      <a
        class="flex h-10 w-10 items-center justify-center text-sm border rounded-l-md hover:bg-gray-100"
        href="undefined?query=test&page=1"
      >
        1
      </a>
      <a
        class="flex h-10 w-10 items-center justify-center text-sm border hover:bg-gray-100"
        href="undefined?query=test&page=2"
      >
        2
      </a>
      <div
        class="flex h-10 w-10 items-center justify-center text-sm border rounded-r-md z-10 bg-blue-600 border-blue-600 text-white"
      >
        3
      </div>
    </div>
    <div
      class="flex h-10 w-10 items-center justify-center rounded-md border pointer-events-none text-gray-300 ml-2 md:ml-4"
    >
      <svg
        aria-hidden="true"
        class="w-4"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.5 4.5 21 12m0 0-7.5 7.5M21 12H3"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </div>
</div>
`;
